src/runtime_src/core/pcie/linux/pcidev.h

```diff
@@ -142,43 +129,93 @@ class pci_device
-  virtual int pcieBarWrite(uint64_t offset, const void *buf, uint64_t len);
-
-  virtual int open(const std::string& subdev, int flag);
-  virtual int open(const std::string& subdev, uint32_t idx, int flag);
-  void close(int devhdl);
-  int ioctl(int devhdl, unsigned long cmd, void *arg = nullptr);
-  int poll(int devhdl, short events, int timeoutMilliSec);
-  virtual void *mmap(int devhdl, size_t len, int prot, int flags, off_t offset);
-  virtual int munmap(int devhdl, void* addr, size_t len);
-  int flock(int devhdl, int op);
-  int get_partinfo(std::vector<std::string>& info, void *blob = nullptr);
-  std::shared_ptr<pcidev::pci_device> lookup_peer_dev();
+  get_subdev_path(const std::string& subdev, uint32_t idx) const;
+
+  virtual int
+  pcieBarRead(uint64_t offset, void *buf, uint64_t len);
+
+  virtual int
+  pcieBarWrite(uint64_t offset, const void *buf, uint64_t len);
+
+  virtual int
+  open(const std::string& subdev, int flag) const;
+
+  virtual int
+  open(const std::string& subdev, uint32_t idx, int flag) const;
+
+  virtual void
+  close(int devhdl) const;
+
+  virtual int
+  ioctl(int devhdl, unsigned long cmd, void *arg = nullptr) const;
+
+  virtual int
+  poll(int devhdl, short events, int timeoutMilliSec);
+
+  virtual void
+  *mmap(int devhdl, size_t len, int prot, int flags, off_t offset);
+
+  virtual int
+  munmap(int devhdl, void* addr, size_t len);
+
+  virtual int
+  flock(int devhdl, int op);
+
+  virtual int
+  get_partinfo(std::vector<std::string>& info, void *blob = nullptr);
+
+  virtual std::shared_ptr<dev>
+  lookup_peer_dev();
+  
+  // Hand out a "device" instance that is specific to this type of device.
+  // Caller will use this device to access device specific implementation of ishim.
+  virtual std::shared_ptr<device>
+  create_device(device::handle_type handle, device::id_type id) const;
+
+  // Hand out an opaque "shim" handle that is specific to this type of device.
+  // On legacy Alveo device, this handle can be used to lookup a device instance and
+  // make xcl HAL API calls.
+  // On new platforms, this handle can only be used to look up a device. HAL API calls
+  // through it are not supported any more.
+  virtual device::handle_type
+  create_shim(device::id_type id) const;
 
 private:
-  int map_usr_bar(void);
+  int
+  map_usr_bar();
+
   std::mutex lock;
   char *user_bar_map = reinterpret_cast<char *>(MAP_FAILED);
-  bool mgmt = false;
 };
 
-void rescan(void);
-size_t get_dev_total(bool user = true);
-size_t get_dev_ready(bool user = true);
-std::shared_ptr<pci_device> get_dev(unsigned index, bool user = true);
+size_t
+get_dev_total(bool user = true);
+
+size_t
+get_dev_ready(bool user = true);
+
+std::shared_ptr<dev>
+get_dev(unsigned index, bool user = true);
+
+int
+get_axlf_section(const std::string& filename, int kind, std::shared_ptr<char>& buf);
+
+int
+get_uuids(std::shared_ptr<char>& dtbbuf, std::vector<std::string>& uuids);
```

Comment:
Looks like these functions are returning status?   If so change to exception and return what is being asked for.  shared_ptr<char> does not look right, maybe std::vector<char>?  What do these  function actually do and where is they used?

Reply:
I have no idea what they are doing and I prefer not to touch them. I merely followed your suggestion and push function name to the very left.

Reply:
That's how I noticed them :-)  Sure leave, but someone has to take a look one day.

Reply:
They won't be implemented in newly derived AIE class anyway :-).

Reply:
It turned out that they are not even used. I'll just remove them completely.